### PR TITLE
[RFC] Load-adjust timer tests

### PR DESCRIPTION
**Pull request by mhinz**
- switch to Xcode 10.1
- calculate load and adjust timer tests
- fix `Test_help_tagjump()`

### CODE REVIEW

**Comment by justinmk** [comment]
can this be closed?

**Comment by justinmk** [comment]
If someone wants to do that, sure. It's not a one-time task: more tests will fail over time, and those need to be monitored, and later fixed.

**Comment by mhinz** [comment]
Then I try looking into the issues of this PR (after I had a look how you handled similar issues earlier).

File: src/nvim/testdir/test_lambda.vim [hunk 1]
```diff
@@ -31,15 +31,13 @@ function! Test_lambda_with_timer()
   endfunction
 
   call s:Foo()
-  sleep 210ms
-  " do not collect lambda
-  call test_garbagecollect_now()
```
Comment (justinmk) [inline src/nvim/testdir/test_lambda.vim]: why remove this? it is part of the upstream test.

Comment (mhinz) [reply src/nvim/testdir/test_lambda.vim]: Happened when I played around and removed all potential error sources. Will add back.

**Comment by mhinz** [comment]
I think this PR is ready now.  💪 

(The failing macOS jobs are due to the unrelated `Test_help_tagjump()` old test.)

File: src/nvim/testdir/test_timers.vim [hunk 2]
```diff
@@ -14,13 +15,17 @@ func MyHandlerWithLists(lists, timer)
...
+endfunc
+
 func Test_oneshot()
   let g:val = 0
   let timer = timer_start(50, 'MyHandler')
   let slept = WaitFor('g:val == 1')
   call assert_equal(1, g:val)
   if has('reltime')
-    call assert_inrange(40, 120, slept)
+    call s:assert_inrange(40, 120, slept)
```
Comment (justinmk) [inline src/nvim/testdir/test_timers.vim]: would be clearer if LoadAdjust() was explicitly called here

Comment (mhinz) [reply src/nvim/testdir/test_timers.vim]: Yup, but you merged it already. :P I guess we just silently do it the next time that file is changed in any way.

**Pull request merged**